validation/gpo-params_test.go

```diff
@@ -0,0 +1,184 @@
+
+		var desiredParams BedrockGasPriceOracleParams
+		switch chain.Superchain {
+		case "mainnet":
+			desiredParams = OPMainnetBedrockGasPriceOracleParams
+		case "goerli":
+			desiredParams = OPGoerliBedrockGasPriceOracleParams
+		case "sepolia":
+			desiredParams = OPSepoliaBedrockGasPriceOracleParams
+		case "goerli-dev-0":
+			t.Fatalf("no ground truth for superchain %s", chain.Superchain)
+		case "sepolia-dev-0":
+			t.Fatalf("no ground truth for superchain %s", chain.Superchain)
+		default:
+			t.Fatalf("superchain not recognized: %s", chain.Superchain)
+		}
+
+		actualParams, err := getBedrockGasPriceOracleParams(context.Background(), gasPriceOraclAddr, client)
+		require.NoError(t, err)
+
+		require.Equal(t, actualParams.Decimals.Cmp(desiredParams.Decimals), 0,
+			"incorrect decimals parameter: got %d, wanted %d", actualParams.Decimals, desiredParams.Decimals)
+		require.Equal(t, actualParams.Overhead.Cmp(desiredParams.Overhead), 0,
+			"incorrect overhead parameter: got %d, wanted %d", actualParams.Overhead, desiredParams.Overhead)
+		require.Equal(t, actualParams.Scalar.Cmp(desiredParams.Scalar), 0,
+			"incorrect scalar parameter: got %d, wanted %d", actualParams.Scalar, desiredParams.Scalar)
+
+		t.Logf("gas price oracle params are acceptable")
+
+	}
+
+	checkEcotoneResourceConfig := func(t *testing.T, chain *ChainConfig, client *ethclient.Client) {
+
+		var desiredParams EcotoneGasPriceOracleParams
+		switch chain.Superchain {
+		case "mainnet":
+			desiredParams = OPMainnetEcotoneGasPriceOracleParams
+		case "goerli":
+			desiredParams = OPGoerliEcotoneGasPriceOracleParams
+		case "sepolia":
+			desiredParams = OPSepoliaEcotoneGasPriceOracleParams
+		case "goerli-dev-0":
+			t.Fatalf("no ground truth for superchain %s", chain.Superchain)
+		case "sepolia-dev-0":
+			t.Fatalf("no ground truth for superchain %s", chain.Superchain)
+		default:
+			t.Fatalf("superchain not recognized: %s", chain.Superchain)
+		}
+
+		actualParams, err := getEcotoneGasPriceOracleParams(context.Background(), gasPriceOraclAddr, client)
+		require.NoError(t, err)
+
+		require.Equal(t, actualParams.Decimals.Cmp(desiredParams.Decimals), 0,
+			"incorrect decimals parameter: got %d, wanted %d", actualParams.Decimals, desiredParams.Decimals)
+		require.Equal(t, actualParams.BlobBaseFeeScalar, desiredParams.BlobBaseFeeScalar,
+			"incorrect blobBaseFeeScalar parameter: got %d, wanted %d", actualParams.BlobBaseFeeScalar, desiredParams.BlobBaseFeeScalar)
+		require.Equal(t, actualParams.BaseFeeScalar, desiredParams.BaseFeeScalar,
+			"incorrect baseFeeScalar: got %d, wanted %d", actualParams.BaseFeeScalar, desiredParams.BaseFeeScalar)
+
+		t.Logf("gas price oracle params are acceptable")
+
+	}
+
+	checkResourceConfig := func(t *testing.T, chain *ChainConfig, client *ethclient.Client) {
+		if Superchains[chain.Superchain].IsEcotone() {
+			checkEcotoneResourceConfig(t, chain, client)
+		}
+		checkBedrockResourceConfig(t, chain, client)
```

Comment:
I've used "bedrock" in a lot of places where I really mean "pre-ecotone". Is that a good name, or not?

Reply:
I'd prefer `preEcotone` since there are other forks in there.